### PR TITLE
init nvram

### PR DESCRIPTION
**Pull request by alokmenghrajani**
Add option to initialize nvram
    
Subzero now supports proper ACLs on the nvram. The default tools
provided by nCipher don't allow setting the ACLs we need, so we need
some helper code.

### CODE REVIEW

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 1]
```diff
@@ -72,6 +118,127 @@ public static void main(String[] args) throws Exception {
...
+      M_Reply rep = sw.getConnection().transact(m_command);
+      if (rep.status != M_Status.OK) {
+        throw new StatusNotOK(
+            "NVMemAlloc returned status " + NFUtils.errorString(rep.status, rep.errorinfo));
+      }
+
+      // Write initial value.
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      baos.write(format("%d-%d", Constants.MAGIC, Constants.VERSION).getBytes(UTF_8));
+      for (int i = 0; i < 92; i++) {
```
Comment (wtmcc) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java]: Should be 100 - amount of data written

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 2]
```diff
@@ -72,6 +118,127 @@ public static void main(String[] args) throws Exception {
...
+          }),
+      };
+      permissionGroups[0].set_certifier(dataKey);
+      permissionGroups[1].set_certifier(acsCert.certs[0].keyhash);
+
+      M_ACL acl = new M_ACL(permissionGroups);
+      M_Cmd_Args_NVMemAlloc args = new M_Cmd_Args_NVMemAlloc(new M_ModuleID(1), 0, fileInfo, acl);
+      M_Command m_command = new M_Command(M_Cmd.NVMemAlloc, 0, args);
+      m_command.set_certs(acsCert);
+      M_Reply rep = sw.getConnection().transact(m_command);
```
Comment (wtmcc) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java]: Use xact?

File: core/src/ncipher/no_rollback.c [hunk 3]
```diff
@@ -14,27 +14,20 @@
...
- * Creating a fresh NVRAM requires an ACS quorum.
- *
- * The process to create the NVRAM and setup the ACLs will live in the Java gui app.
+ * Creating a fresh NVRAM requires an ACS quorum. The easiest way to setup the
+ * NVRAM is to use the Java GUI application with --init-nvram.
  *
  * Note: we decided not to use the INCR instruction. We feel it's overly complicated.
  *
- * When testing, the following commands can be used to delete or allocate the NVRAM.
+ * For testing purpose, the following commands can be used to delete or allocate the NVRAM.
```
[the remark this answers is not in the document]
Comment (alokmenghrajani) [reply core/src/ncipher/no_rollback.c]: fixed

File: java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java [hunk 4]
```diff
@@ -125,6 +125,21 @@ public void promptForOperatorCard(String message) throws IOException {
...
+   * Prompt to insert an Administrator Card.  Returns once the card is inserted.
+   *
+   * @param message Message to display to the user
+   */
+  public void promptForAdministratorCard(String message) throws IOException {
+    framebuffer.draw((Graphics2D graphics) ->
+        baseScreenLayout(graphics,"Insert Administrator Card", message, Color.white)
+    );
+
+    // TODO: Wait on the Operator Card instead
```
[the remark this answers is not in the document]
Comment (alokmenghrajani) [reply java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java]: fixed

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 5]
```diff
@@ -72,6 +118,127 @@ public static void main(String[] args) throws Exception {
...
+    // Load the ACS
+    M_CertificateList acsCert = buildAcsCert(sw);
+
+    // Get data signer key hash
+    Key datasigner_key = sw.getKey("seeinteg", config.dataSignerKey);
+
+    M_KeyHash dataKey = toKeyHash(datasigner_key.getData().hash);
+
+    // Create both NVRAM files
+    for (String file : new String[] { "selfcheck\00\00", "subzero00\00\00"}) {
```
[the remark this answers is not in the document]
Comment (alokmenghrajani) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java]: fixed

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java (hunk 1 above)
Comment (alokmenghrajani) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java]: fixed

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java (hunk 2 above)
Comment (alokmenghrajani) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java]: fixed

**Pull request merged**